src/lib/world/src/importing.rs

```diff
@@ -1,172 +1,221 @@
+    fn get_chunk_count(&self, import_dir: &Path) -> Result<u64, WorldError> {
         info!("Counting chunks in import directory...");
         let regions_dir = import_dir.join("region").read_dir()?;
         let chunk_count = AtomicU64::new(0);
+
         regions_dir
-            .into_iter()
             .par_bridge()
-            .for_each(|region_file| match region_file {
-                Ok(dir_entry) => {
-                    if dir_entry.path().is_dir() {
-                        error!(
-                            "Region file is a directory: {}",
-                            dir_entry.path().to_string_lossy()
-                        );
-                    } else {
-                        let file_path = dir_entry.path();
-                        let Ok(anvil_file) = load_anvil_file(file_path.clone()) else {
-                            error!(
-                                "Could not load region file: {}",
-                                file_path.clone().display()
-                            );
-                            return;
-                        };
-                        let locations = anvil_file.get_locations();
-                        chunk_count.fetch_add(
-                            locations.len() as u64,
-                            std::sync::atomic::Ordering::Relaxed,
-                        );
-                    }
+            .try_for_each(|region_file| -> Result<(), WorldError> {
+                let entry = region_file?;
+                if entry.path().is_dir() {
+                    return Ok(());
                 }
-                Err(e) => {
-                    error!("Could not read region file: {}", e);
+
+                if let Ok(anvil_file) = load_anvil_file(entry.path()) {
+                    chunk_count
+                        .fetch_add(anvil_file.get_locations().len() as u64, Ordering::Relaxed);
                 }
-            });
-        Ok(chunk_count.load(std::sync::atomic::Ordering::Relaxed))
+                Ok(())
+            })?;
+
+        Ok(chunk_count.load(Ordering::Relaxed))
     }
 
     pub async fn import(&mut self, import_dir: PathBuf, _: PathBuf) -> Result<(), WorldError> {
-        // Check if the import path is valid. We can assume the database path is valid since we
-        // checked it in the config validity check.
-        check_paths_validity(import_dir.clone())?;
-        let regions_dir = import_dir.join("region").read_dir()?;
-        let progress_bar = Arc::new(ProgressBar::new(self.get_chunk_count(import_dir)?));
-        info!("Importing chunks from import directory...");
-        let start = std::time::Instant::now();
-        let mut task_set = JoinSet::new();
+        check_paths_validity(&import_dir)?;
+
+        let total_chunks = self.get_chunk_count(&import_dir)?;
+        let progress_style = ProgressStyle::default_bar()
+            .template("[{elapsed_precise}] {bar:40.cyan/blue} {pos:>7}/{len:7} {msg}")
+            .unwrap();
```

Comment:
don't use unwrap

Reply:
i have to? don't i?

Reply:
no? Print out the error with a description or return it if the function returns a result

Reply:
this is the progress  bar dude